File: ruby/udemy_class/arrays/reverse_sentence.rb
def reverse_sentence(sentence)
  words = []
  current_word = ''

  sentence.split('').each do |char|
    if char == ' '
      words << current_word
      current_word = ''
    else
      current_word += char
    end
  end

  words << current_word

  index = words.length - 1

  while index > 0
    print "#{words[index]} "
    index -= 1
  end

  print "#{words[0]}."
end

reverse_sentence('This is the best')

File: python/udemy_class/arrays/reverse_sentence.py
def reverse_sentence(sentence):
    words = []
    current_word = ""

    i = 0

    while i < len(sentence):
        if sentence[i] == " ":
            words.append(current_word)
            current_word = " "
        else:
            current_word += sentence[i]

        i += 1

    words.append(current_word)

    i = len(words) - 1

    while i > 0:
        print(f"{words[i]} ")

        i -= 1

    print(f"{words[0]}.")

reverse_sentence("This is the best.")
